### PR TITLE
psp-3986 cache list of lookups.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: source/frontend/src/customAxios.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export const CustomAxios = ({
     baseURL,
     headers: {
       'Access-Control-Allow-Origin': '*',
+      'Cache-Control': process.env.NODE_ENV === 'development' ? 'no-cache' : '',
```
Comment (devinleighsmith) [inline source/frontend/src/customAxios.ts:45]: In development, don't cache anything as we may be changing lookup codes around a bunch.

File: source/backend/api/Controllers/LookupController.cs [hunk 2]
```diff
@@ -76,102 +83,112 @@ public IActionResult GetPropertyClassificationTypes()
...
-            codes.AddRange(documentStatusTypes);
-            codes.AddRange(acquisitionFileStatusTypes);
-            codes.AddRange(acquisitionPhysFileStatusTypes);
-            codes.AddRange(acquisitionTypes);
-            codes.AddRange(activityTemplateTypes);
-            codes.AddRange(activityStatusTypes);
-            codes.AddRange(acqFilePersonProfileTypes);
-            codes.AddRange(tenantTypes);
-            codes.AddRange(acqFundingTypes);
+            if (!_memoryCache.TryGetValue(CacheKeys.Lookup, out JsonResult cachedLookupResponse))
```
Comment (devinleighsmith) [inline source/backend/api/Controllers/LookupController.cs:90]: Store this result in memory, this should make all future requests for the lookup codes very fast and reduce need for db transactions until the server is restarted.

File: source/backend/api/Controllers/LookupController.cs [hunk 3]
```diff
@@ -76,102 +83,112 @@ public IActionResult GetPropertyClassificationTypes()
         [Produces("application/json")]
         [ProducesResponseType(typeof(IEnumerable<Model.LookupModel>), 200)]
         [SwaggerOperation(Tags = new[] { "lookup" })]
+        [ResponseCache(Duration = OneHourInSeconds)]
```
Comment (devinleighsmith) [inline source/backend/api/Controllers/LookupController.cs:86]: Could use some feedback here. My worry is that a user could: 1) Log into the app 2) get a list of lookup codes 3) we do a database update 4) the list of codes is now out of date.

The risk of this in prod seem low but I can see this happening in our lower environments. Perhaps this is more trouble then its worth?

File: source/backend/api/Controllers/LookupController.cs [hunk 4]
```diff
@@ -76,102 +83,112 @@ public IActionResult GetPropertyClassificationTypes()
...
+                codes.AddRange(acquisitionTypes);
+                codes.AddRange(activityTemplateTypes);
+                codes.AddRange(activityStatusTypes);
+                codes.AddRange(acqFilePersonProfileTypes);
+                codes.AddRange(tenantTypes);
+                codes.AddRange(acqFundingTypes);
+
+                var response = new JsonResult(codes);
+
+                _memoryCache.Set(CacheKeys.Lookup, response);
```
Comment (devinleighsmith) [inline source/backend/api/Controllers/LookupController.cs:187]: store the result in memory if not already present.

File: source/backend/api/Controllers/LookupController.cs (hunk 3 above)
Comment (FuriousLlama) [reply source/backend/api/Controllers/LookupController.cs:86]: It really depends on how often the codes get updated. AFAIK they require a migration, so not very often. I think it should be ok to cache them for an hour.

**Review by FuriousLlama — APPROVED** [review]

**Review by asanchezr — APPROVED** [review]

**Pull request merged**